integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/SparkCommonIT.java

```diff
@@ -509,6 +509,35 @@ void testAlterTableUpdateColumnComment() {
     checkTableColumns(tableName, updateCommentColumns, getTableInfo(tableName));
   }
 
+  @Test
+  void testAlterTableReplaceColumns() {
+    String tableName = "test_replace_columns_table";
+    dropTableIfExists(tableName);
+
+    createSimpleTable(tableName);
+    List<SparkColumnInfo> simpleTableColumns = getSimpleTableColumn();
+    checkTableColumns(tableName, simpleTableColumns, getTableInfo(tableName));
+
+    //  We must make type compatibility when replace columns to pass Hive IT. Or Will get such
+    // error: Failed to alter Hive table [test_replace_columns_table] in Hive metastore, since Hive
+    // metastore will check the compatibility of column type between the old and new column
+    // positions, please ensure that the type of the new column position is compatible with the old
+    // one, otherwise the alter operation will fail in Hive metastore.
+    sql(
+        String.format(
+            "ALTER TABLE %S REPLACE COLUMNS (id int COMMENT 'new comment', name2 string, age long);",
+            tableName));
+    ArrayList<SparkColumnInfo> updateColumns = new ArrayList<>();
+    // change comment for id
+    updateColumns.add(SparkColumnInfo.of("id", DataTypes.IntegerType, "new comment"));
+    // change column name
+    updateColumns.add(SparkColumnInfo.of("name2", DataTypes.StringType, null));
+    // change column type
+    updateColumns.add(SparkColumnInfo.of("age", DataTypes.LongType, null));
+
+    checkTableColumns(tableName, updateColumns, getTableInfo(tableName));
```

Comment:
Could you please test the table data read/write before and after altering it? 

I am worried that the table may not be read or written properly after being modified.

Reply:
ok

Reply:
updated